include/nbl/builtin/hlsl/sort/counting.hlsl

```diff
@@ -33,97 +33,98 @@ struct counting
                 template __call <SharedAccessor>(value, sdata);
     }
 
-    void build_histogram(NBL_REF_ARG( KeyAccessor) key, NBL_REF_ARG(SharedAccessor) sdata, const CountingParameters<Key> data)
+    uint32_t toroidal_histogram_add(uint32_t tid, uint32_t sum, NBL_REF_ARG(HistogramAccessor) histogram, NBL_REF_ARG(SharedAccessor) sdata, const CountingParameters<Key> params)
+    {
+        sdata.workgroupExecutionAndMemoryBarrier();
+
+        sdata.set(tid % GroupSize, sum);
+        uint32_t shifted_tid = (tid + glsl::gl_SubgroupSize() * params.workGroupIndex) % GroupSize;
+
+        sdata.workgroupExecutionAndMemoryBarrier();
```

Comment:
why are you shifting around it shared memory!?

Its literally enough to do
```cpp

````

Reply:
ok lets go back to the original simple `histogram.atomicAdd(tid,sdata.get(tid));`

I forgot the prefix sum is not already in smem, so the extra barriers will just make us slower.

Reply:
Removed